lib/Sema/SemaExpr.cpp

```diff
@@ -11935,7 +11935,20 @@ ExprResult Sema::ActOnCountBoundsExpr(SourceLocation BoundsKWLoc,
                                       CountBoundsExpr::Kind Kind,
                                       Expr *CountExpr,
                                       SourceLocation RParenLoc) {
-  return new (Context) CountBoundsExpr(Kind, CountExpr, BoundsKWLoc, RParenLoc);
+  ExprResult Result = UsualUnaryConversions(CountExpr);
```

Comment:
>UsualUnaryConversions [](start = 22, length = 21)

Is this a standard canonicalization?  A comment or two in this function explaining the general approach would be helpful.